### ISSUE TITLE
Filters are applied backward

### ISSUE THREAD

**Issue by atomrc** [staltz/xstream#85]
When chaining `filter`s on a stream, they are executed backward at run time. 

eg
```javascript
const filtered$ = value$ 
    .filter(val => val.foo)
    //this filter is played before the one above ... causing an error
    .filter(val => val.foo.bar === "ok")
```

and here is a jsbin to reproduce the error: https://jsbin.com/noribadoxa/edit?html,js,console

**Comment by staltz** [comment]
Thanks for reporting.

**Comment by staltz** [comment]
Fixed in v5.2.4.

**Comment by atomrc** [comment]
Awesome thanks :)
